packages/react-sdk/src/wrappers/LivestreamPlayer/LivestreamPlayer.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react';
+import { Call } from '@stream-io/video-client';
+import { useStreamVideoClient } from '@stream-io/video-react-bindings';
+import {
+  LivestreamLayout,
+  LivestreamLayoutProps,
+  StreamCall,
+} from '../../core';
+
+export type LivestreamPlayerProps = {
+  /**
+   * The call type. Usually `livestream`.
+   */
+  callType: string;
+  /**
+   * The call ID.
+   */
+  callId: string;
+  /**
+   * The props for the {@link LivestreamLayout} component.
+   */
+  layoutProps?: LivestreamLayoutProps;
+};
+
+export const LivestreamPlayer = (props: LivestreamPlayerProps) => {
+  const { callType, callId, layoutProps } = props;
+  const client = useStreamVideoClient();
+  const [call, setCall] = useState<Call>();
+  useEffect(() => {
+    if (!client) return;
+    const myCall = client.call(callType, callId);
+    setCall(myCall);
+    myCall.join().catch((e) => {
+      console.error('Failed to join call', e);
+    });
+    return () => {
+      myCall.leave().catch((e) => {
```

Comment:
Sorry for neat-picking but I believe try-catch would be better here rather than .catch, since this is the pattern we follow most of the places in the code.

Reply:
hey @khushal87, we can't go without `.catch()` inside a `useEffect()`. This is because `useEffect` doesn't work with async code that well. Consider the following code:
```tsx
useEffect(() => {
  const effect = async () => {
    try { 
      await doStuff(); 
    } catch (e) { 
      handle(e); 
    }
  }
  effect();
}, []);
```

Although now `effect` wraps everything in a `try/catch` block, it still needs to have a `.catch()` cause to handle the possible future rejections. So, we would have to implement this effect like this:

```tsx
useEffect(() => {
  const effect = async () => {
    try { 
      await doStuff(); 
    } catch (e) { 
      handle(e); 
    }
  }
  effect().catch((err) => handle(err));
}, []);
```

Ultimately, we can simplify to:
```tsx
useEffect(() => {
  doStuff().catch((err) => handle(err));
}, []);
```